### PR TITLE
added Entities.findbyName

### PR DESCRIPTION
**Pull request by MiladNazeri**
Adding feature to find entities by name as I've seen a lot of users do this kind of thing in scripts:

```
 Entities.findEntities(MyAvatar.position, SEARCH_RADIUS).forEach(function(element){
            var name = Entities.getEntityProperties(element, 'name').name;
		
            if (name === "Balloon Cage"){
                _this.balloonCage = element;
                print("found my balloonCage");
                // print(element);
                // print(name);
                return;
            }

        });
		
        if (_this.balloonCage === undefined){
            print("No Balloon Cage Found!");
        }			
```
This just cuts out having to do the for loop to see if those entities match by name

You can do the above now with: 
Entities.findEntitiesByName("Ballon Cage", MyAvatar.position, 30, true);

test plan:

1. create a box named "box-test"
2. open console and run Entities.findEntitiesByName("box-test", MyAvatar.position, 30, false);
3. check the results back is the id of box-test
4. create another box named "Box-Test"
5. run the same command in the console and see that you get back an array with two ids.
6. run Entities.findEntitiesByName("box-test", MyAvatar.position, 30, true);
7. Confirm that you only match the box with proper case

### CODE REVIEW

**Comment by MiladNazeri** [comment]
https://highfidelity.fogbugz.com/f/cases/13977/Scripting-Feature-Added-Entities-findEntityByName

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/8527/HighFidelity-Beta-PR12796-4adc8928f09cf1b02949b76a3a3157849659668a.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/8527/HighFidelity-Beta-PR12796-4adc8928f09cf1b02949b76a3a3157849659668a.exe

File: libraries/entities/src/EntityScriptingInterface.cpp [hunk 1]
```diff
@@ -734,6 +734,24 @@ QVector<QUuid> EntityScriptingInterface::findEntitiesByType(const QString entity
     return result;
 }
 
+QVector<QUuid> EntityScriptingInterface::findEntitiesByName(const QString entityName, const glm::vec3& center, float radius) const {
```
Comment (sethalves) [inline libraries/entities/src/EntityScriptingInterface.cpp]: Couldn't this just be a utility in a javascript library?

File: tools/jsdoc/package-lock.json [hunk 2]
```diff
@@ -0,0 +1,138 @@
+{
```
Comment (sethalves) [inline tools/jsdoc/package-lock.json]: was this file accidentally added?

File: libraries/entities/src/EntityScriptingInterface.cpp (hunk 1 above)
Comment (MiladNazeri) [reply libraries/entities/src/EntityScriptingInterface.cpp]: I've seen people write this code over and over again doing findEntities, and checking each one one to match a name to grab an ID.  I can close this PR if you are against it, but think this is much more convenient for users in practice.  

Didn't realize those other two files snuck in there, if you want to keep this pr, I'll remove those right away.

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/8737/app_12796_PR_3443c112acac5fa66c3835858b53c1819787953e.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/8737/HighFidelity-Beta-PR12796-3443c112acac5fa66c3835858b53c1819787953e.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/8737/HighFidelity-Beta-Interface-PR12796-3443c112acac5fa66c3835858b53c1819787953e.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/8737/HighFidelity-Beta-PR12796-3443c112acac5fa66c3835858b53c1819787953e.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/8737/HighFidelity-Beta-Interface-PR12796-3443c112acac5fa66c3835858b53c1819787953e.exe

File: libraries/entities/src/EntityScriptingInterface.h [hunk 3]
```diff
@@ -387,6 +387,20 @@ public slots:
     /// this function will not find any entities in script engine contexts which don't have access to entities
     Q_INVOKABLE QVector<QUuid> findEntitiesByType(const QString entityType, const glm::vec3& center, float radius) const;
 
+    /**jsdoc
+    * Find all entities of a particular name that intersect a sphere defined by a center point and radius.
+    * @function Entities.findEntitiesByName
+    * @param {Entities.EntityType} entityName - The name of the entity to search for.
```
Comment (ctrlaltdavid) [inline libraries/entities/src/EntityScriptingInterface.h]: Perhaps add, "Case sensitive." to the end of the entityName description?

Comment (MiladNazeri) [reply libraries/entities/src/EntityScriptingInterface.h]: Good point. 

Do you think that is the best route to go, or would it be better to add a param for bool on case-sensitive search?

**Comment by MiladNazeri** [comment]
Added a case sensitive search bool param

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/8843/app_12796_PR_988dca437d6b753cfbae11f88bb32b699e19d428.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/8843/HighFidelity-Beta-PR12796-988dca437d6b753cfbae11f88bb32b699e19d428.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/8843/HighFidelity-Beta-Interface-PR12796-988dca437d6b753cfbae11f88bb32b699e19d428.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/8843/HighFidelity-Beta-PR12796-988dca437d6b753cfbae11f88bb32b699e19d428.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/8843/HighFidelity-Beta-Interface-PR12796-988dca437d6b753cfbae11f88bb32b699e19d428.exe

**Comment by misslivirose** [comment]
I like this change -- while it can be accomplished in JS, I think as long as we have the other types of findBy functions in this API, we should include findByName

**Comment by qlimnavarro** [comment]
Pass @conklin94122

**Comment by qlimnavarro** [comment]
Currently Testing

**Comment by qlimnavarro** [comment]
Pass @conklin94122 
IDs appear successfuly and more than 1 appear when there are more boxes.

**Pull request merged**